Modules/_sre.c

```diff
@@ -2308,10 +2308,19 @@ match_repr(MatchObject *self)
     PyObject *group0 = match_getslice_by_index(self, 0, Py_None);
     if (group0 == NULL)
         return NULL;
-    result = PyUnicode_FromFormat(
-            "<%s object; span=(%zd, %zd), match=%.50R>",
-            Py_TYPE(self)->tp_name,
-            self->mark[0], self->mark[1], group0);
+
+    /* Truncate group0 with ... if repr will be longer than 50 characters */
+    if (PySequence_Length(group0) <= 48 - PyBytes_Check(group0)) {
```

Comment:
This makes some implicit assumptions about repr (that for string it will add 2 characters and for bytes 3 characters)

Reply:
That's a correct assumption:

```
>>> a = '"\''
>>> len(a)
2
>>> len(repr(a))
5
>>> b = b'\x00'
>>> len(b)
1
>>> len(repr(b))
7
```
I think the best way will be to use `reprlib` which will make it easy to both limit the output to 50 characters and add the closing quote (otherwise you would have to find whether to use `'` or `"`):

```
>>> import reprlib
>>> r = reprlib.Repr()
>>> a = 'abcde'*10
>>> print(r.repr(a))
'abcdeabcdeab...cdeabcdeabcde'
```